### DISCUSSION TITLE
How to filter work items returned by type

### DISCUSSION THREAD

**Discussion by oranjutanj** [rfennell/AzurePipelines#1294] [Q&A] [answered]
First of all absolutely LOVE this extension, game and indeed life changer.

I have it working just the way I want, but I'm struggling with combining the handlebar syntax with how to logically only display what I need.

My question is, _how can I filter the work items I return by work item type_? 

I don't want to include tasks in the list. I have the standard section below:
```
|ID|Title|Type|State|Area Path|
|--|--|--|--|--|
{{#forEach this.workItems}} 
| [**{{this.id}}**]({{lookup this._links.html 'href'}}) | {{lookup this.fields 'System.Title'}} | {{lookup this.fields 'System.WorkItemType'}} | {{lookup this.fields 'System.State'}} | {{lookup this.fields 'System.AreaPath'}} |
{{/forEach}}
```

But I want to not include a row if 'System.WorkItemType' == "Task". I'm guessing you can either specify in the query somehow, or there is some way to include a logical operator by using extensions. There's lots of examples about how to do string comparisons on the internet like this one https://stackoverflow.com/questions/34252817/handlebarsjs-check-if-a-string-is-equal-to-a-value but they didn't work when I translated them into the required syntax (I think), like this:

```
module.exports = 
{compareStrings(p, q, options) {
    return (p == q) ? options.fn(this) : options.inverse(this);
}
};
```

and then in the template

```
{{#forEach this.workItems}} 
{{#with this.fields}}
{{#compareStrings 'System.WorkItemType' "Task"}}
**Worky! {{'System.WorkItemType'}}**
{{else}}
No worky
{{/compareStrings}}
{{/with}}
{{/forEach}}`
```
But all I get is "no worky" :)

Can anyone think of another way to do this, or something I'm missing?

Many thanks,
Mike

**Comment by rfennell** [comment, marked as answer]
You should be able to do it all within the template, no extra extension needed. Something like

```
|ID|Title|Type|State|Area Path|
|--|--|--|--|--|
{{#forEach this.workItems}} 
{{#if (eq (lookup this.fields 'System.WorkItemType') 'Bug')}}
| [**{{this.id}}**]({{lookup this._links.html 'href'}}) | {{lookup this.fields 'System.Title'}} | {{lookup this.fields 'System.WorkItemType'}} | {{lookup this.fields 'System.State'}} | {{lookup this.fields 'System.AreaPath'}} |
{{/if}}
{{/forEach}}
```

The comparison is done using one of the [helper functions](https://github.com/helpers/handlebars-helpers) I include

**Reply by oranjutanj** [reply]
Thanks so much, sorry I was too unfamiliar with how the helper methods worked to figure that out! That works a treat thank you (or rather I just put the line in the {{else}} block).
